src/kit.ts

```diff
@@ -642,8 +703,11 @@ export async function scanForVSKits(pr?: ProgressReporter): Promise<Kit[]> {
   const installs = await vsInstallations();
   const prs = installs.map(async(inst): Promise<Kit[]> => {
     const ret = [] as Kit[];
-    const arches = ['x86', 'amd64', 'x86_amd64', 'x86_arm', 'amd64_arm', 'amd64_x86'];
-    const sub_prs = arches.map(arch => tryCreateNewVCEnvironment(inst, arch, pr));
+    // TODO: exclude toolsets that are not yet installed.
```

Comment:
Doesn't `tryCreateNewVCEnvironment` handle your TODO?  I thought it would only succeed if the tools were there and skip it otherwise.

Reply:
It appears that vcvarsall.bat, when invoked for arm/arm64 on an install that does not include these architectures, it still creates a valid environment for x86. To me this is confusing, but I can follow up separately with the VS team, in case there is a good reason for this. The environment difference between the two cases is that for arm(64) installed, the first path is [hostArch]\[targetArch] instead of [hostArch]\[hostArch]. I am fixing it now by checking on this detail.